### PR TITLE
#6921 JUnit 4 to JUnit 5 Migration of all the examples

### PR DESCRIPTION
**Pull request by nagkumar**
#6921

Pl. check the @Rule and @RuleChain Migrations as I am  not sure if that is perfect, I avoided ExtendsWith as it was not so independent work

Also, make common jars come to junit4.gradle to make changes min in the future for such upgrade.

Kindly give your feedback, so that I can destroy my fork once this request is accepted and merged.

### CODE REVIEW

File: examples/build.gradle [hunk 1]
```diff
@@ -6,8 +6,10 @@ plugins {
 apply from: "$rootDir/../gradle/ci-support.gradle"
 
 subprojects {
-    apply plugin:"java"
+    apply plugin: "java"
```
Comment (eddumelendez) [inline examples/build.gradle:9]: let's avoid those kind of changes, please.

File: examples/build.gradle [hunk 2]
```diff
@@ -18,4 +20,7 @@ subprojects {
         toolVersion = "9.3"
         configFile = rootProject.file('../config/checkstyle/checkstyle.xml')
     }
+
+    task prepareKotlinBuildScriptModel {
```
Comment (eddumelendez) [inline examples/build.gradle:24]: is this really needed?

File: examples/cucumber/build.gradle [hunk 3]
```diff
@@ -10,8 +10,10 @@ dependencies {
...
     implementation 'org.seleniumhq.selenium:selenium-chrome-driver:3.141.59'
-    testImplementation 'io.cucumber:cucumber-java:7.11.1'
-    testImplementation 'io.cucumber:cucumber-junit:7.11.1'
     testImplementation 'org.testcontainers:selenium'
-    testImplementation 'org.assertj:assertj-core:3.24.2'
+
+    testImplementation(platform("io.cucumber:cucumber-bom:7.12.0"))
+    testImplementation("io.cucumber:cucumber-java")
+    testImplementation("io.cucumber:cucumber-junit-platform-engine")
+    testImplementation 'io.cucumber:cucumber-junit:7.12.0'
```
Comment (eddumelendez) [inline examples/cucumber/build.gradle:18]: if the bom is imported we can remove the version
```suggestion
    testImplementation 'io.cucumber:cucumber-junit'
```

File: examples/docs/ref/output.txt [hunk 4]
```diff
@@ -0,0 +1,26 @@
+C:\temp\ramu\testcontainers-java\examples>gradle clean test
```
Comment (eddumelendez) [inline examples/docs/ref/output.txt:1]: remove this file

File: examples/hazelcast/build.gradle [hunk 5]
```diff
@@ -7,8 +7,6 @@ repositories {
 }
 
 dependencies {
-    testImplementation 'org.testcontainers:testcontainers'
     testImplementation 'com.hazelcast:hazelcast:5.2.3'
     testImplementation 'ch.qos.logback:logback-classic:1.3.5'
-    testImplementation 'org.assertj:assertj-core:3.24.2'
```
Comment (eddumelendez) [inline examples/hazelcast/build.gradle:13]: Let's avoid this. examples should contain everything is needed to run it.

File: examples/junit5.gradle [hunk 6]
```diff
@@ -0,0 +1,23 @@
+def jjVersion = "5.9.2"
```
Comment (eddumelendez) [inline examples/junit5.gradle:1]: remove this file, please. every project should be as clear as possible.

**Review by eddumelendez — CHANGES_REQUESTED** [review]
Thanks for your contribution and patience. I have left some comments.

**Comment by nagkumar** [comment]
Thank You for the review, I shall raise a new PR with all the fixes, just one clarification to remove junit5.gradle point

The elimination of junit5.gralde would lead to adding all of the junit dependencies in each of the build file, that way when we move from junit 5 to say junit 6 again all these have to be changed in all of the examples, i.e. currently 17 examples exist, all the Gradle files have to change again to upgrade junit version.. 

If you are u ok with such repeat and error-prone (as they are so independent, some examples may still be using junit 5 and junit 6 etc..actually it **is already happening now while all the examples are junit 4, while neo4j-container is already junit 5** )

```
    testImplementation(platform("org.junit:junit-bom:$jjVersion"))
    testImplementation("org.junit.jupiter:junit-jupiter")
    testImplementation("org.junit.platform:junit-platform-suite")

    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine")
    testRuntimeOnly('org.junit.vintage:junit-vintage-engine')
```

image

Please advise on this one point alone to share my revised PR.

**Comment by eddumelendez** [comment]
new PR is not needed. Just update the branch used for this PR

**Comment by nagkumar** [comment]
Just saw your message.. after I created a new PR.. hope you don't mind.. in all new code is the same repo.. but did do a fresh changes now.

**Pull request closed without merging**